tests/Aspire.Workload.Tests/PerTestFrameworkTemplatesTests.cs

```diff
@@ -0,0 +1,179 @@
+
+        res = await cmd.ExecuteAsync($"test -c {config}");
+
+        Assert.True(res.ExitCode != 0, $"Expected the tests project build to fail");
+        Assert.Matches("System.ArgumentException.*Resource 'webfrontend' not found.", res.Output);
+        Assert.Matches("Failed! * - Failed: *1, Passed: *0, Skipped: *0, Total: *1", res.Output);
+
+        async Task AssertBasicTemplateAsync(IBrowserContext context)
+        {
+            await project.StartAppHostAsync(extraArgs: [$"-c {config}"]);
+
+            try
+            {
+                var page = await project.OpenDashboardPageAsync(context);
+                await CheckDashboardHasResourcesAsync(page, []);
+            }
+            finally
+            {
+                await project.StopAppHostAsync();
+            }
+        }
+
+        static void PrepareTestProject(AspireProject project, string projectPath)
+        {
+            // Insert <ProjectReference Include="$(MSBuildThisFileDirectory)..\aspire-starter0.AppHost\aspire-starter0.AppHost.csproj" /> in the project file
+
+            // taken from https://raw.githubusercontent.com/dotnet/templating/a325ffa18edd1590f9b340cf83d51d8eb567ebdc/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ValueForms/XmlEncodeValueFormFactory.cs
+            StringBuilder output = new();
+            using (var w = XmlWriter.Create(output, s_xmlWriterSettings))
+            {
+                w.WriteString(project.Id);
+            }
+            var xmlEncodedId = output.ToString();
+
+            var projectReference = $@"<ProjectReference Include=""$(MSBuildThisFileDirectory)..\{xmlEncodedId}.AppHost\{xmlEncodedId}.AppHost.csproj"" />";
+
+            var newContents = File.ReadAllText(projectPath)
+                                    .Replace("</Project>", $"<ItemGroup>{projectReference}</ItemGroup>\n</Project>");
+            File.WriteAllText(projectPath, newContents);
+        }
+
+        static void PrepareTestCsFile(string id, string projectDir, string testTemplateName)
+        {
+            var testCsPath = Path.Combine(projectDir, "IntegrationTest1.cs");
+            var sb = new StringBuilder();
+
+            // Uncomment everything after the marker line
+            var inTest = false;
+            var marker = testTemplateName switch
+            {
+                "aspire-nunit" => "// [Test]",
+                "aspire-mstest" => "// [TestMethod]",
+                "aspire-xunit" => "// [Fact]",
+                _ => throw new NotImplementedException($"Unknown test template: {testTemplateName}")
+            };
+
+            foreach (var line in File.ReadAllLines(testCsPath))
+            {
+                if (!inTest && line.Contains(marker))
+                {
+                    inTest = true;
+                }
+
+                if (inTest && CommentLineRegex().IsMatch(line))
+                {
+                    sb.AppendLine(CommentLineRegex().Replace(line, "    "));
```

Comment:
Hopefully we don't ever add real comments to the code, or else this would start failing.

Reply:
Lol! Yeah, but that would cause a failure on the PR, and then that person gets to figure it out :/